### PR TITLE
Add sorting to all resources

### PR DESCRIPTION
**Pull request by kookster**
implements #145

### CODE REVIEW

**Comment by kookster** [comment]
finally finished this one up, yeesh @cavis 
I'll schedule review time with @sandikbarr

File: app/controllers/api/auth/stories_controller.rb [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 class Api::Auth::StoriesController < Api::StoriesController
-
   include ApiAuthenticated
 
   api_versions :v1
 
   filter_resources_by :account_id, :series_id, :network_id
 
-  filter_params :noseries
+  filter_params :highlighted, :purchased, :v4, :text, :noseries
```
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: Also, it's slightly strange to define the `filter_params` here, when it's the filtered `super(resources)` call that actually applies them.

File: app/controllers/api/stories_controller.rb [hunk 2]
```diff
@@ -74,11 +77,8 @@ def scoped(relation)
   end
 
   def sorted(res)
-    if filters.purchased?
-      res.purchased.order('purchase_count DESC')
-    else
-      res.order('published_at desc')
-    end
+    res = res.purchased.order('purchase_count DESC') if filters.purchased?
```
Comment (cavis) [inline app/controllers/api/stories_controller.rb]: Just to double check ... this `order` will get applied before the default sort, correct?

File: test/models/tagging_test.rb [hunk 3]
```diff
@@ -15,6 +15,10 @@
     tagging.must_respond_to(:taggable)
   end
 
+  it 'has an owner' do
+    tagging.owner.wont_be_nil
```
Comment (cavis) [inline test/models/tagging_test.rb]: Off-topic?

Comment (kookster) [inline test/models/tagging_test.rb]: yes it absolutely is, I just noticed some missing unit tests in the code coverage and put that in as a bonus.

File: app/controllers/api/auth/stories_controller.rb (hunk 1 above)
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: Yeah, this is probably the more explicit option, then.  So be it!

File: app/controllers/api/auth/stories_controller.rb [hunk 4]
```diff
@@ -20,18 +19,14 @@ def check_user_network
     user_not_authorized unless current_user.networks.exists?(params[:network_id])
   end
 
-  def sorted(res)
-    res.order('updated_at desc')
```
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: I think that may be a good default for the API.  When viewing all stories ... you get them in order of published date.  When viewing only your stories (including unpublished), you get them in order of updated_at.

File: app/controllers/api/auth/stories_controller.rb [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
   include ApiAuthenticated
 
   api_versions :v1
 
   filter_resources_by :account_id, :series_id, :network_id
 
-  filter_params :noseries
+  filter_params :highlighted, :purchased, :v4, :text, :noseries
+
+  sort_params default: { updated_at: :desc },
```
Comment (kookster) [inline app/controllers/api/auth/stories_controller.rb]: default is now updated_at @cavis

File: app/controllers/api/auth/stories_controller.rb (hunk 4 above)
Comment (kookster) [inline app/controllers/api/auth/stories_controller.rb]: fixed above.

File: app/controllers/api/auth/stories_controller.rb (hunk 5 above)
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: Nice.  And it still inherits `allowed` from the parent?

Comment (kookster) [inline app/controllers/api/auth/stories_controller.rb]: no, when I call `sort_params`, I have to specify both, so line 13 has the allowed sort params.
(I didn't implement partial overrides.)

Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: Gotcha.

**Comment by cavis** [comment]
Looks good to me.  Unassigning myself!

File: app/controllers/concerns/api_sorting.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  private
+
+  def parse_sorts_param
+    sorts_array = []
+    allowed_sorts = self.class.allowed_sort_names || self.class.superclass.allowed_sort_names
+
+    # parse sort param for name of the column and direction
+    # default is descending, because I say so, and we have a bias towards the new
+    (params[:sorts] || '').split(',').each do |str|
```
Comment (sandikbarr) [inline app/controllers/concerns/api_sorting.rb]: add a comment example of this: /?sorts=published_at:desc,updated_at:desc

**Pull request merged**